src/cunumeric/index/repeat.cu

```diff
@@ -0,0 +1,164 @@
+    const size_t offset = (idx * gridDim.x + blockIdx.x) * blockDim.x + threadIdx.x;
+    if (offset < volume) {
+      auto point      = pitches.unflatten(offset, origin);
+      auto val        = repeats[point];
+      offsets[offset] = val;
+      SumReduction<int64_t>::fold<true>(value, val);
+    }
+  }
+  // Every thread in the thread block must participate in the exchange to get correct results
+  reduce_output(out, value);
+}
+
+template <typename VAL, int DIM>
+__global__ static void __launch_bounds__(THREADS_PER_BLOCK, MIN_CTAS_PER_SM)
+  repeat_kernel(Buffer<VAL> out,
+                const AccessorRO<VAL, DIM> in,
+                int64_t repeats,
+                const int32_t axis,
+                const Rect<DIM> rect,
+                const Pitches<DIM - 1> pitches,
+                const int volume)
+{
+  const size_t idx = blockIdx.x * blockDim.x + threadIdx.x;
+  if (idx >= volume) return;
+  int64_t i = idx / repeats;
+  auto p    = pitches.unflatten(i, rect.lo);
+  out[idx]  = in[p];
+}
+
+template <typename VAL, int DIM>
+__global__ static void __launch_bounds__(THREADS_PER_BLOCK, MIN_CTAS_PER_SM)
+  repeat_kernel(Buffer<VAL> out,
+                const AccessorRO<VAL, DIM> in,
+                const AccessorRO<int64_t, DIM> repeats,
+                Buffer<int64_t> offsets,
+                const int32_t axis,
+                const Rect<DIM> rect,
+                const Pitches<DIM - 1> pitches,
+                const int volume)
+{
+  const size_t idx = blockIdx.x * blockDim.x + threadIdx.x;
+  if (idx >= volume) return;
+  auto p         = pitches.unflatten(idx, rect.lo);
+  size_t out_idx = offsets[idx];
+  for (size_t r = 0; r < repeats[p]; r++) {
+    out[out_idx] = in[p];
+    ++out_idx;
+  }
+}
+
+template <LegateTypeCode CODE, int DIM>
+struct RepeatImplBody<VariantKind::GPU, CODE, DIM> {
+  using VAL = legate_type_of<CODE>;
+
+  size_t operator()(Buffer<VAL>& out,
+                    const AccessorRO<VAL, DIM>& in,
+                    const int64_t repeats,
+                    const int32_t axis,
+                    const Pitches<DIM - 1>& pitches,
+                    const Rect<DIM>& rect) const
+  {
+    const size_t volume = rect.volume();
+    size_t size         = volume * repeats;
+    const size_t blocks = (size + THREADS_PER_BLOCK - 1) / THREADS_PER_BLOCK;
+
+    out = create_buffer<VAL>(size, Memory::Kind::GPU_FB_MEM);
+
+    repeat_kernel<VAL, DIM>
+      <<<blocks, THREADS_PER_BLOCK>>>(out, in, repeats, axis, rect, pitches, size);
+    return size;
+  }
+
+  size_t operator()(Buffer<VAL>& out,
+                    const AccessorRO<VAL, DIM>& in,
+                    const AccessorRO<int64_t, DIM>& repeats,
+                    const int32_t axis,
+                    const Pitches<DIM - 1>& pitches,
+                    const Rect<DIM>& rect) const
+  {
+    const size_t volume = rect.volume();
+
+    cudaStream_t stream;
+    cudaStreamCreate(&stream);
```

Comment:
Use get_cached_stream instead of creating a new stream.
https://github.com/nv-legate/cunumeric/blob/branch-22.03/src/cunumeric/cuda_help.h#L110